modules/obam/manifests/obam_au.pp

```diff
@@ -0,0 +1,126 @@
+#----------------------------------------------------------------------------
+#  Copyright (c) 2020 WSO2, Inc. http://www.wso2.org
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#----------------------------------------------------------------------------
+
+class obam::obam_au inherits obam::params{
+
+include stdlib
+
+if ($spec =="AU")  {
+notify{"Executing Spec ${spec} for OBAM":}
+#OBAM Config Edits for AU Spec
+
+file_line {'editing AU spec PrimaryAuthenticator, open-banking.xml' :
+  path  => $obam_openbanking ,
+  line   => '<SCA><PrimaryAuthenticator> <Name>IdentifierExecutor</Name> <DisplayName>identifier-first</DisplayName> </PrimaryAuthenticator></SCA>',
+  after => '<Server>',
+  multiple => false,
+  match => '<SCA><PrimaryAuthenticator> <Name>IdentifierExecutor</Name> <DisplayName>identifier-first</DisplayName> </PrimaryAuthenticator></SCA>',
```

Comment:
Match is used to replace the config if found. Why have we used the same line to be replaced?

Reply:
As i found out match function can be used to replace if the specific tags exists or if not exists append.

Reply:
Shouldn't we use append_on_no_match to append?. The issue I see here is after is used to add the line after the specified line and match is used to replace/append according to append_on_no_match config. Why have we used both here?

Reply:
As i found out append_on_no_match is already default set true. I added it because without a match, it will not check if existing tags are there and there will add a duplicate entry.